packages/admin-panel/src/layout/AppPageLayout.jsx

```diff
@@ -2,25 +2,83 @@
  * Tupaia
  *  Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
  */
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import { IconButton } from '@material-ui/core';
 import { Outlet } from 'react-router-dom';
+import styled from 'styled-components';
 import { labelToId } from '../utilities';
-import { Main, PageWrapper } from './Page';
 import { NavPanel } from './navigation';
+import { CaretLeftIcon } from '../icons';
+import { NAV_PANEL_CLOSED_WIDTH, NAV_PANEL_OPEN_WIDTH } from './navigation/NavPanel';
 
-export const AppPageLayout = ({ user, routes, logo, homeLink, userLinks, basePath }) => {
+const PageWrapper = styled.div`
+  display: flex;
+`;
+
+const Main = styled.main`
+  overflow-x: auto;
+  height: 100vh;
+  // This is so that we can make the PageBody component fill the whole remaining height of the screen
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+`;
+
+const ArrowButton = styled(IconButton)`
+  position: absolute;
+  top: 1.8rem;
+  right: ${props => (props.$navOpen ? '0.2rem' : '-1rem')};
+  padding-block: 0.5rem;
+  padding-inline: ${props =>
+    props.$navOpen
+      ? '0.5rem 0.55rem'
+      : '0.55rem 0.5rem'}; // make up for the visual effect of the icon point
+  background-color: ${props => props.theme.palette.secondary.main};
+  color: ${props => props.theme.palette.common.white};
+  z-index: 1201; // above the drawer
+  transition: right 0.2s;
```

Comment:
Nice touch 💃

Could perhaps include `transform` too?

```suggestion
  transition: right 0.2s, transform 0.2s;
```

Alternatively:

```suggestion
  transition-duration: 200ms;
  transition-property: right, transform;
  transition-timing-function: ease;
```